internal/gomod/dependencies.go

```diff
@@ -95,15 +120,88 @@ func parseGoListOutput(output, rootVersion string) (map[string]Module, error) {
 			module.Version = rootVersion
 		}
 
-		dependencies[importPath] = Module{
+		dependencies[importPath] = GoModule{
 			Name:    module.Name,
 			Version: cleanVersion(module.Version),
 		}
 	}
 
+	if goVersion.Version == "" {
+		return nil, errors.New("Must have a valid go version in go mod file")
+	}
+	setGolangDependency(dependencies, goVersion)
+
 	return dependencies, nil
 }
 
+type parsedGoVersion struct {
+	Version string
+	Major   int
+	Minor   int
+}
```

Comment:
We use a semver package in sg/sg that might be useful here as well. Saves you from parsing the version manually here.